packages/nextra/src/compile.ts

```diff
@@ -91,7 +92,14 @@ export async function compileMdx(
     ]
   })
   try {
-    const vFile = await compiler.process(source)
+    const vFile = await compiler.process(
+      filePath
+        ? {
+            value: source,
+            path: filePath
+          }
+        : source
+    )
```

Comment:
According to https://github.com/orgs/mdx-js/discussions/2219#discussioncomment-4649484 it's good to also include the filepath for plugins (if they need it).